Source/WebKit/UIProcess/AboutSchemeHandler.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "AboutSchemeHandler.h"
+
+#if USE(APPLE_INTERNAL_SDK) // && __has_include(<WebKitAdditions/AboutSchemeHandlerAdditions.h>)
+#include <WebKitAdditions/AboutSchemeHandlerAdditions.h>
+#endif
+
+namespace WebKit {
+
+AboutSchemeHandler& AboutSchemeHandler::singleton()
+{
+    static MainThreadNeverDestroyed<AboutSchemeHandler> globalHandler;
+    return globalHandler;
+}
+
+AboutSchemeHandler::AboutSchemeHandler()
+{
+    platformInitialize();
+}
+
+void AboutSchemeHandler::platformStartTask(WebPageProxy&, WebURLSchemeTask& task)
+{
+    auto requestURL = task.request().url();
+    if (requestURL.hasOpaquePath()) {
+        auto opaquePath = requestURL.path();
+
+        if (auto it = m_handlers.find(opaquePath.toStringWithoutCopying()); it != m_handlers.end()) {
```

Comment:
Can use `find<StringViewHashTranslator>(opaquePath)`

Reply:
Also, why don't we use `get()` and null check the result?

Reply:
Because I didn't know that! Thanks!